agent/api/container/container.go

```diff
@@ -781,6 +787,92 @@ func (c *Container) BuildContainerDependency(contName string,
 	c.TransitionDependenciesMap[dependentStatus] = deps
 }
 
+// GetResources returns the list of container resources from ResourcesMap
+func (c *Container) GetResources() []taskresource.TaskResource {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	return c.getResourcesUnsafe()
+}
+
+// getResourcesUnsafe returns the list of task resources from ResourcesMap
+func (c *Container) getResourcesUnsafe() []taskresource.TaskResource {
+	var resourceList []taskresource.TaskResource
+	for _, resources := range c.ResourcesMapUnsafe {
+		resourceList = append(resourceList, resources...)
+	}
+	return resourceList
+}
+
+// AddResource adds a resource to ResourcesMap
+func (c *Container) AddResource(resourceType string, resource taskresource.TaskResource) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.ResourcesMapUnsafe[resourceType] = append(c.ResourcesMapUnsafe[resourceType], resource)
+}
+
+// GetASMAuthResource retrieves asmauth resource from resource map
+func (c *Container) GetASMAuthResource() ([]taskresource.TaskResource, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	res, ok := c.ResourcesMapUnsafe[asmauth.ResourceName]
+	return res, ok
+}
+
+// GetSSMSecretsResource retrieves ssmsecret resource from resource map
+func (c *Container) GetSSMSecretsResource() ([]taskresource.TaskResource, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	res, ok := c.ResourcesMapUnsafe[ssmsecret.ResourceName]
+	return res, ok
+}
+
+// GetASMSecretsResource retrieves ssmsecret resource from resource map
+func (c *Container) GetASMSecretsResource() ([]taskresource.TaskResource, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	res, ok := c.ResourcesMapUnsafe[asmsecret.ResourceName]
+	return res, ok
+}
+
+// GetSSMSecretRequirements retrieves ASMauth data requirements of the container
+func (c *Container) GetASMAuthDataRequirements() []*containerresource.ASMAuthData {
+	var reqs []*containerresource.ASMAuthData
+	reqs = append(reqs, c.RegistryAuthentication.ASMAuthData)
+	return reqs
+}
+
+// GetSSMSecretRequirements retrieves container secrets in a map whose key is region and value is all
+// secrets in that region
+func (c *Container) GetSSMSecretRequirements() map[string][]containerresource.Secret {
+	reqs := make(map[string][]containerresource.Secret)
+	for _, secret := range c.Secrets {
+		if secret.Provider == SecretProviderSSM {
+			if _, ok := reqs[secret.Region]; !ok {
+				reqs[secret.Region] = []containerresource.Secret{}
+			}
+			reqs[secret.Region] = append(reqs[secret.Region], secret)
```

Comment:
Do you need a `break` here?

Reply:
Hmmm we will need to collect all the SSM secrets for the container I don't think we can have `break` statement here..